test/unit/active_record_cursor_test.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+require "test_helper"
+
+module JobIteration
+  class ActiveRecordCursorTest < IterationUnitTest
+    test "#next_batch preloads the following batch asynchronously" do
+      skip("Only supported on Rails >= 7") unless Product.connection.try(:async_enabled?)
+
+      cursor = ActiveRecordCursor.new(Product.all)
+      cursor.next_batch(2)
+
+      assert_predicate(cursor.next_relation, :scheduled?)
```

Comment:
On the fence about whether `cursor` should have an attribute reader for `:next_relation` - it makes testing easier, but perhaps this should be kept private as internal state.

We could also verify the async loading behaviour with mocha:

```
ActiveRecord::Relation.any_instance.expects(:load_async)
```

But this felt awkward too in it's own way...